### PR TITLE
codestyle: add type hints to elchecking module

### PR DESCRIPTION
**Pull request by THS-on**
- codestyle: add type hint to tpm class constructor
- codestyle: add type hints to elchecking module

### CODE REVIEW

File: keylime/elchecking/tests.py [hunk 1]
```diff
@@ -10,11 +10,13 @@
 # A Test can be used multiple times, even concurrently.
 
 # Data is the type of Python data that corresponds to JSON values.
-Data = typing.Union[int, float, str, bool, typing.Tuple["Data", ...], typing.Mapping[str, "Data"], None]
+Data = typing.Union[
+    int, float, str, bool, typing.Tuple["Data", ...], typing.Mapping[str, "Data"], typing.List["Data"], None
```
Comment (aplanas) [inline keylime/elchecking/tests.py:14]: Here and a bit later in line 19 we refer again the ABC instead the type. Is there a real advantage?

**Review by aplanas — APPROVED** [review]

File: mypy.ini [hunk 2]
```diff
@@ -10,5 +10,9 @@ ignore_errors = False
 [mypy-keylime.config]
 ignore_errors = False
 
+[mypy-keylime.elchecking.*]
+ignore_errors = False
+
+
```
[the remark this answers is not in the document]
Comment (THS-on) [reply mypy.ini]: Nope, fixed.

File: keylime/elchecking/tests.py (hunk 1 above)
Comment (THS-on) [reply keylime/elchecking/tests.py:14]: I don't know. What would be your suggestion?

File: keylime/elchecking/policies.py [hunk 3]
```diff
@@ -52,20 +52,20 @@ class RejectAll(Policy):
...
     def get_relevant_pcrs(self) -> typing.FrozenSet[int]:
-        return set()
+        return frozenset()
 
     def refstate_to_test(self, refstate: RefState) -> tests.Test:
         return tests.RejectAll("reject all")
 
 
-def _mkreg() -> typing.Mapping[str, Policy]:
+def _mkreg() -> typing.MutableMapping[str, Policy]:
```
[the remark this answers is not in the document]
Comment (aplanas) [reply keylime/elchecking/policies.py:61]: IMHO we should use the lowest type possible. Python is not like Go, it is not structural typing, so I can implement an object that implement the same functions than `MutableMapping` and still not be a `MuatableMapping`.

There is also the issue with coherence. Until now seems that we are following this rule (favor `Dict` instead the interface).

But in any case it is OK for me : ) :+1:

**Review by aplanas — APPROVED** [review]

File: keylime/elchecking/policies.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (THS-on) [reply keylime/elchecking/policies.py:61]: Ok, that makes sense I'll change that.

**Review by aplanas — APPROVED** [review]

**Review by aplanas — APPROVED** [review]

**Pull request merged**